sdk/metric/exemplar/histogram_reservoir.go

```diff
@@ -12,13 +12,21 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 )
 
+// HistogramReservoirProvider is a provider of [HistogramReservoir].
+func HistogramReservoirProvider(bounds []float64) ReservoirProvider {
+	cp := slices.Clone(bounds)
+	slices.Sort(cp)
+	return func(_ attribute.Set) Reservoir {
+		return NewHistogramReservoir(cp)
+	}
+}
+
 // NewHistogramReservoir returns a [HistogramReservoir] that samples the last
 // measurement that falls within a histogram bucket. The histogram bucket
 // upper-boundaries are define by bounds.
 //
-// The passed bounds will be sorted by this function.
+// The passed bounds must be sorted before calling this function.
 func NewHistogramReservoir(bounds []float64) *HistogramReservoir {
-	slices.Sort(bounds)
```

Comment:
It is.  I can leave this as-is if we are not comfortable with this change (see https://github.com/open-telemetry/opentelemetry-go/pull/5861#issuecomment-2414575828).  It is just a small performance penalty to sort the bounds each time we create a reservoir (once per attribute-set), rather than sort each time we create a reservoir provider (once per aggregation) if we keep this behavior as-is.